ckan-2.10/base/Dockerfile.py3.10

```diff
@@ -69,19 +69,31 @@ RUN pip3 install -U pip && \
     ckan config-tool ${CKAN_INI} "beaker.session.secret = " && \
     ckan config-tool ${CKAN_INI} "ckan.plugins = ${CKAN__PLUGINS}"
 
-# Create a local user and group plus set up the storage path
-RUN groupadd -g 92 ckan && \
-    useradd -rm -d /srv/app -s /bin/bash -g ckan -u 92 ckan && \
-    mkdir -p ${CKAN_STORAGE_PATH} && \
-    chown -R ckan:ckan ${CKAN_STORAGE_PATH}
-
+# Create ckan and ckan-sys users and the ckan-sys group plus set up the storage path
+RUN groupadd -g 502 ckan-sys && \
+    useradd -rm -d /srv/app -s /bin/bash -g ckan-sys -u 502 ckan-sys && \
+    useradd -rm -d /srv/app -s /bin/bash -g ckan-sys -u 503 ckan
+    
 COPY setup/prerun.py ${APP_DIR}
 COPY setup/start_ckan.sh ${APP_DIR}
 ADD https://raw.githubusercontent.com/ckan/ckan/${CKAN_VERSION}/wsgi.py ${APP_DIR}
 RUN chmod 644 ${APP_DIR}/wsgi.py
 
 # Create entrypoint directory for children image scripts
-ONBUILD RUN mkdir /docker-entrypoint.d
+RUN mkdir -p /docker-entrypoint.d && chmod 755 /docker-entrypoint.d
+
+# Set the ownership of the app directory, usr/local and the entrypoint directory to the ckan-sys user
+RUN chown -R ckan-sys:ckan-sys ${APP_DIR} && \
+    chown -R ckan-sys:ckan-sys /docker-entrypoint.d && \
+    chown -R ckan-sys:ckan-sys /usr/local
+
+# Set the ownership of the CKAN config file, src and the storage path to the ckan user
```

Comment:
```suggestion
# Set the ownership of the CKAN config file, src and the storage path to the ckan-sys user
```

Reply:
I needed to set ownership of the `ckan.ini` file to the`ckan` user (which runs the processes) as the `prerun.py` script runs a plugin update. Another alternative could be to set the **group** ownership of the `ckan.ini` file to `ckan-sys` as this is the primary group for both users...